src/common.jl

```diff
@@ -7,3 +7,37 @@ end
 macro _dimcheck(condition)
     :(($(esc(condition))) || throw(DimensionMismatch("Dimensions of the parameters don't match")))
 end
+
+"""
+A helper macro to create a const (submod) which mimics a submodule containing the symbols from an export expression (expr).
+
+```
+module A
+type X end
+type Y end
+@autocomplete SubMod export X,Y
+end
+
+using A
+
+# now you can type A.SubMod.<tab> to autocomplete X and Y
+```
+"""
+macro autocomplete(submod::Symbol, expr::Expr)
```

Comment:
@Evizero How do you feel about using this macro as an alternative to submodules?  I think it will give you the autocomplete functionality you were going for, but keeping the code structure simpler.